python/MooseDocs/extensions/sqa.py

```diff
@@ -234,3 +262,286 @@ def __contains__(self, name):
+                    settings = MooseMarkdownCommon.getSettingsHelper(options,
+                                                                     match.group('settings'),
+                                                                     legacy_style=False)
+                    title = settings['title']
+                    for item in match.group('items').split('\n'):
+                        tag_id, text = re.split(r'\s+', item.strip(), maxsplit=1)
+                        out[title].append((tag_id, text))
+        return out
+
+    def buildText(self, key, filename, tag_id, text):
+        """
+        Create the content for the collapsible body
+        """
+        if tag_id not in self._database[key]:
+            LOG.error('Unknown tag: %s', tag_id)
+            tag = '<span class="moose-sqa-error">{}</span>'.format(tag_id)
+            desc = '<span class="moose-sqa-error">{}</span>'.format(text)
+            body = desc
+        else:
+            desc = '<div class="moose-sqa-list-description">{}</div>'.format(text)
+            body = desc
+            tag = '<a href="{}">{}</a>'.format(filename, tag_id)
+            for item in self._database[key][tag_id]:
+                body += '<div class="moose-sqa-list-link"><a href="{}">{}</a></div>'.format(*item)
+        return tag, desc, body
+
+class SQAInputTagDatabase(object):
+    """
+    Storage container for SQA tagged groups (e.g., requirements).
+
+    This storage is used for recall by the traceability matrix.
+    """
+    RE = re.compile(r'@(?P<key>\w+)\s+(?P<tag>.*?)(?:$|\s+)', flags=re.IGNORECASE)
+
+    def __init__(self, repo):
+        self.__repo = repo
+        self.__database = collections.defaultdict(lambda: collections.defaultdict(list))
+        for base, _, files in os.walk(MooseDocs.ROOT_DIR):
+            for filename in files:
+                full_name = os.path.join(base, filename)
+                if filename.endswith('.i'):
+                    self.search(full_name)
+
+    def __getitem__(self, value):
+        """
+        Operator[] access to the top-level of the storage dict.
+        """
+        return self.__database[value]
+
+    def __contains__(self, key):
+        """
+        Keyword "in" access to the top-level of the storage dict.
+        """
+        return key in self.__database
+
+    def search(self, filename):
+        """
+        Locates @<keyword> tags within input file.
+        """
+        with open(filename) as fid:
+            for match in self.RE.finditer(fid.read()):
+                key = match.group('key').lower()
+                tag = match.group('tag')
+                local = filename.replace(MooseDocs.ROOT_DIR, '').lstrip('/')
+                remote = '{}/{}'.format(self.__repo.rstrip('/'), local)
+                self.__database[key][tag].append((remote, local))
+
+class SQAPageStatus(MooseMarkdownCommon, Pattern):
+    """
+    Adds input tag matrix creation (e.g., Requirements Traceability Matrix)
```

Comment:
Is this what this class does? It seems it is the same text as for `SQAInputTagMatrix`